src/backend/distributed/executor/multi_client_executor.c

```diff
@@ -469,6 +469,8 @@ MultiClientQueryResult(int32 connectionId, void **queryResult, int *rowCount,
 	{
 		WarnRemoteError(connection, result);
 		PQclear(result);
+
+		return false;
```

Comment:
(please evaluate and feel free to overlook)

Echoing @marcocitus' note here. libpq has the most leaky API ever and I remember that we had to drain result objects from the connection in certain cases (and not do that for certain other ``PQresultStatus()`` statuses)

If we think we don't need ``ClearRemainingResults()``, what if we moved that call and ``return true`` into the ``if`` block?